src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -0,0 +1,112 @@
+            DurableTaskExtension config,
+            HttpClient httpClientFactory)
+        {
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.httpClient = httpClientFactory;
+        }
+
+        public override string Run(TaskContext context, string input)
+        {
+            // This won't get called as long as we've implemented RunAsync.
+            throw new NotImplementedException();
+        }
+
+        public async override Task<string> RunAsync(TaskContext context, string rawInput)
+        {
+            HttpRequestMessage requestMessage = await CreateHttpRequestMessageAsync(rawInput);
+            HttpResponseMessage response = await this.httpClient.SendAsync(requestMessage);
+            DurableHttpResponse durableHttpResponse = await CreateDurableHttpResponseAsync(response);
+
+            string serializedOutput = MessagePayloadDataConverter.HttpConverter.Serialize(durableHttpResponse, true);
+            return serializedOutput;
+        }
+
+        private static async Task<DurableHttpResponse> CreateDurableHttpResponseAsync(HttpResponseMessage response)
+        {
+            DurableHttpResponse durableHttpResponse = new DurableHttpResponse(response.StatusCode);
+            durableHttpResponse.Headers = CreateStringValuesHeaderDictionary(response.Headers);
+            durableHttpResponse.Content = await response.Content.ReadAsStringAsync();
+            return durableHttpResponse;
+        }
+
+        private static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    foreach (var headerValue in header.Value)
+                    {
+                        if (newHeaders.ContainsKey(header.Key))
+                        {
+                            StringValues values = StringValues.Concat(headerValue, newHeaders[header.Key]);
+                            newHeaders[header.Key] = values;
+                        }
+                        else
+                        {
+                            StringValues stringValues = new StringValues(headerValue);
+                            newHeaders.Add(header.Key, stringValues);
+                        }
+                    }
+                }
+            }
+
+            return newHeaders;
+        }
+
+        private static async Task<HttpRequestMessage> CreateHttpRequestMessageAsync(string rawInput)
+        {
+            JsonSerializerSettings serializerSettings = new JsonSerializerSettings();
+            serializerSettings.TypeNameHandling = TypeNameHandling.Auto;
+
+            IList<string> input = JsonConvert.DeserializeObject<List<string>>(rawInput, serializerSettings);
```

Comment:
Consider adding a comment about why the value of `rawInput` is a `List<string>`, as opposed to just a `DurableHttpRequest`. You may remember the details now, but that is something you are likely to forget in a month or two.